### DISCUSSION TITLE
we need an output of type Date

### DISCUSSION THREAD

**Discussion by SMMirshojaee** [SaeedDev94/ng-persian-datepicker#61] [Ideas]
hi. thank you for this brilliant package.
i think anybody use ur package, needs to access `Date` type of selected date. `ngModel` doesn't work and the only object we have access, is value of a `FormControl` as string.
i tried to get datepicker element with `@ViewChild` and see properties inside it. there is a `selectedDate` that is unfortunately private and a `dateValue` that is private too 😞 .
if there is any way, i'm here to listen and use it.
with best regards... 🌷

**Comment by SaeedDev94** [comment]
Hi
The lib has 2 major events:

1- dateOnInit: IActiveDate
2- dateOnSelect: IActiveDate

Since `IActiveDate` including timestamp you can easily create `Date` object

**Reply by SaeedDev94** [reply]
@SMMirshojaee You can checkout `IActiveDate` here:
**[IActiveDate](https://github.com/Saeed-Pooyanfar/ng-persian-datepicker/blob/master/projects/ng-persian-datepicker/src/lib/interface/IActiveDate.ts)**

**Reply by SMMirshojaee** [reply]
thank you.
I will be happy if you mention this in document too.

**Reply by SaeedDev94** [reply]
@SMMirshojaee **[README.md: IActiveDate](https://github.com/Saeed-Pooyanfar/ng-persian-datepicker#iactivedate)**

**Reply by SMMirshojaee** [reply]
Perfect
